### PR TITLE
Add preEvaluationHash field to BlockType

### LINKED ISSUE #3280: feat(libplanet-explorer): Support `preEvaluationHash` field in `BlockType`
## Usage

```graphql
// When using LibplanetExplorerSchema
query GetBlockPreEvaluationHash {
  blockQuery {
    block(hash: "<HASH>") {
      preEvaluationHash
    }
  }
}
```

### PR DESCRIPTION
**Pull request by XxshiftxX**
Add `PreEvaluationHash` field to `BlockType`. (resolves #3280)

### CODE REVIEW

**Comment by moreal** [comment]
@XxshiftxX You should update `CHANGES.md`. Maybe, it should place in the *Added APIs*.

File: CHANGES.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ Released on June 28, 2023.
  -  Added `IBlockChainStates.GetState()` interface method.  [[#3250]]
  -  Added `IBlockStates.GetState()` interface method.  [[#3250]]
  -  Changed `IBlockStates` to inherit `IAccountState` interface.  [[#3251]]
+ -  Added `BlockType.PreEvaluationHash` field. [[#3281]]
```
Comment (moreal) [inline CHANGES.md:51]: I guess this pull request's branch is not on the latest commit of the `main` branch. Could you rebase the branch on the `main` branch?

File: CHANGES.md [hunk 2]
```diff
@@ -59,6 +59,7 @@ be compatible with this version.
     `List<IReadOnlyList<string>>?` from `List<List<string>>?`.  [[#3274]]
  -  (Libplanet.Explorer) Changed the type for `TxResult.ActionsLogList` to
     `List<IReadOnlyList<string>>?` from `List<List<string>>?`.  [[#3274]]
+ -  (Libplanet.Explorer) Added `BlockType.PreEvaluationHash` field. [[#3281]]
```
Comment (tkiapril) [inline CHANGES.md:62]:
```suggestion
 -  (Libplanet.Explorer) Added `BlockType.PreEvaluationHash` field. [[#3280], [#3281]]
```
Must mention the resolving issue (or related thereof) as well.

Also, you must add the following to the end of the changelog for this version:
```md
[#3280]: https://github.com/planetarium/libplanet/issues/3280
[#3281]: https://github.com/planetarium/libplanet/pull/3281
````

**Review by tkiapril — CHANGES_REQUESTED** [review]

File: CHANGES.md (hunk 2 above)
Comment (XxshiftxX) [reply CHANGES.md:62]: I just thought it's link doesn't work because it takes some time to sync. Thank you for let me know. 👍

File: CHANGES.md [hunk 3]
```diff
@@ -91,6 +92,8 @@ be compatible with this version.
 [#3272]: https://github.com/planetarium/libplanet/pull/3272
 [#3273]: https://github.com/planetarium/libplanet/pull/3273
 [#3274]: https://github.com/planetarium/libplanet/pull/3274
+[#3280]: https://github.com/planetarium/libplanet/pull/3280
```
Comment (tkiapril) [inline CHANGES.md:95]:
```suggestion
[#3280]: https://github.com/planetarium/libplanet/issues/3280
```

🤭

**Review by tkiapril — CHANGES_REQUESTED** [review]

**Comment by tkiapril** [comment]
Also, I guess we might need some tests here. https://github.com/planetarium/libplanet/blob/main/Libplanet.Explorer.Tests/GraphTypes/BlockTypeTest.cs

Besides, I'm not entirely sure if `preEvaluationHash` belongs to `BlockType`, can anybody give an insight? @planetarium/libplanet

**Review by tkiapril — DISMISSED** [review]

**Review by moreal — APPROVED** [review]

**Review by tkiapril — APPROVED** [review]
Might need to squash and merge.

**Pull request merged**